cli: Hide Watch Usage output when running on non-interactive envs

taylon opened the pull request:
<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. The two fields below are mandatory. -->

<!-- Please remember to update CHANGELOG.md in the root of the project if you have not done so. -->

**Summary**

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
 
Watch Usage options are useless in non-interactive environments (like Docker). Currently jest-cli is smart enough to know that it should not clear the terminal in those environments, with this PR it will also not output Watch Usage related stuff either, making the UI cleaner.

Also, because the terminal is not cleared after a test run, the subsequent runs will not print a new line after the "Press w to show more" text, which makes it hard to read and a really bad UX once you have watch running for a couple of test runs.

I'm not familiar with the code base at all, so let me know if there is anything that should be done differently and I'll fix it =D

**Test plan**

<!-- Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI. -->

Output in interactive environments is not affected.
Here is how the output looks in Docker before and after this PR:

**Before:**
![before](https://user-images.githubusercontent.com/79991/33234516-4125a92a-d20f-11e7-9978-b2590d2aaca0.png)

**After:**
![after](https://user-images.githubusercontent.com/79991/33234519-469ee344-d20f-11e7-8067-6e98ade0b39d.png)

rogeliog commented on packages/jest-cli/src/check_if_interative.js:
```diff
@@ -0,0 +1,3 @@
+import isCI from 'is-ci';
+
+export default process.stdout.isTTY && !isCI;
```
NIT: I think we could name this file `is_interactive` so that the import looks like 'import isInteractive from './is_interactive'...

I could even be nice if this was in something like `jest-util` and then we could use it in other places of the codebase like
```
packages/jest-cli/src/pre_run_message.js
packages/jest-cli/src/reporters/__tests__/default_reporter.test.js
packages/jest-cli/src/reporters/coverage_reporter.js
packages/jest-cli/src/reporters/default_reporter.js
packages/jest-cli/src/watch.js
packages/jest-util/src/clear_line.js
```

rogeliog commented:
Looks good!

One small comment if you are running jest in watch mode within Docker. You could pass the `-it` flags and then it would make it interactive `docker run -it my-image yarn test`

cpojer commented:
Thanks for sending a PR. Can you make sure "yarn test" passes? :)

taylon commented:
I renamed and moved isInteractive to jest-util and also replaced usage of `process.stdout.isTTY && !isCI` with it.

I couldn't figure out why tests are failing in circleci though. Locally I tried `yarn test`, `yarn test-ci` and `yarn test-ci-partial` and everything passes. Do you guys have any idea of what could be wrong?

SimenB commented:
Do they pass when you do `CI=true yarn test`? Or somehow emulate the terminal on CI

taylon commented:
Thanks @SimenB that was exactly it =D

Tests are passing now, I also added tests to is_interactive.
CircleCI is failing because of a yarn error and I can't trigger another build manually, that should be passing though.

SimenB commented:
I triggered a new CI run.

Can you update the changelog?
